src/lib/process/index.ts

```diff
@@ -50,29 +52,47 @@ export const createProcess = async (
   dryRun: boolean = false,
   options: Polkadot.Options = defaultOptions
 ): Promise<Process.Result> => {
-  const program: Process.Program = validate(userProgram)
-  const processId = utf8ToHex(name, Constants.PROCESS_ID_LENGTH)
-  const polkadot: Polkadot.Polkadot = await createNodeApi(options)
-  const expectedVersion: number = (await getVersion(polkadot, processId)) + 1
+  try {
 
-  if (version !== expectedVersion) {
-    throw new VersionError(`Version: ${version} must be incremented to: ${expectedVersion}`)
-  }
+    const program: Process.Program = validate(userProgram)
+    const processId = utf8ToHex(name, Constants.PROCESS_ID_LENGTH)
+    const polkadot: Polkadot.Polkadot = await createNodeApi(options)
+    const currentVersion: number = await getVersion(polkadot, processId)
+    const expectedVersion: number = currentVersion + 1
 
-  if (dryRun)
-    return {
-      process: null,
-      message: 'Dry run: transaction has not been created',
-      name,
-      version: expectedVersion,
-      program,
+    if (version > expectedVersion || version < currentVersion)
+      throw new VersionError(version, expectedVersion, name)
+
+    if (version === currentVersion) {
+      const process = await getProcess(polkadot, processId, version)
+
+      if (program.length !== process.program.length) 
+        throw new ProgramError('existing: programs are different lengths', process) 
+
+      if (!program.every((step, i) => textify(step) === textify(process.program[i])))
+        throw new ProgramError('existing: program steps did not match', process) 
+
+      throw new ProgramError(`Process ${name} is already created.`, process)
```

Comment:
could argue this isn't an error. The user hasn't made an erroor if they're trying to create a process that's exactly the same as one already created, it's a skip. Whereas the other `ProgramError`s are telling the user 'there is something wrong in your input'

Reply:
In a way this is wrong with the input? 

as you are trying to create something that is already created?

Guys, it feels like this is getting frustrating/picky.

Reply:
I think the point of the ticket is that '...trying to create something that is already created?' is OK

Reply:
in the demo scenario, the skip case happens a lot, because the 3-persona docker compose does `npm run flows` every time is starts. If you haven't cleared chain state that means the output will be `...is already created` for every process, which is not a probem -> not an error.

Reply:
so what is your suggestion?

Reply:
Should have been updated to return process along with test scenario.